src/redux/books/books.js

```diff
@@ -0,0 +1,33 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+const initialState = [];
+
+const addRemoveReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        action.book,
+      ];
+    case REMOVE_BOOK:
+      return [
+        ...state.slice(0, state.indexOf(action.book)),
+        ...state.slice(state.indexOf(action.book) + 1, state.length),
```

Comment:
- [ ] Checking again on the requirements of this project, in case the `action.type` is `REMOVE_BOOK`, the function should use the `ES6 filter()` method to create a new array that will not contain the book you want to remove from the store. 
- **Hint**: `state.filter(book => book.id !== id);`. Here, I'm filtering by the id key.
- **NB** Let's avoid using methods that mutate the original array like slice()


![](https://media.giphy.com/media/taDxtc7by09TZ03ciP/giphy.gif)

Reply:
Thank you for your review. Just so you know though, `slice()` doesn't mutate the state array. It only returns a new array from the state array.

Reply:
Hi @Kamba56, this should be an error. You are perfectly correct. I didn't pay attention. I thought it was splice().